### DISCUSSION TITLE
No Power reading unless recording / Prompt for  Strava upload

### DISCUSSION THREAD

**Discussion by medington** [cagnulein/qdomyos-zwift#2400] [Q&A] [answered]
Hi, thank you so much for creating this very useful app. 🙇🏻‍♂️

I've been using it for a couple of years now with My Echelon EX-4S and it works great.

I have question about using the app to warm up. When I start spinning the app shows my heart rate but no power output (or any of the related metrics). I assume this is either a deliberate design or a limitation of some sort. I typically do not want my warm up to be part of my Strava recording so I don't click start until I'm done. This leaves me in the dark about where my power is during that time.

If there were some type of prompt at the end of a session like "Upload to Strava?" then I could just do a throw-away record session for my warm-up. Otherwise I have to go in and delete it in both Strava and Apple health.

Please let me know if I'm missing something or if this is a worthwhile feature request. Since this is open source, I'd even be willing to look into doing the work as I have experience developing iPhone apps. Thanks again!

**Comment by cagnulein** [comment, marked as answer]
Hi, thanks for the feedback!
The fact that the wattage is not calculated during the paused mode was
about a very old modification that i did for a user, i can check if i can
reverse it
also i can add a popup for strava upload request

with the current version you can achieve what you need simply pressing
start and when you are ok with the warmup just close the qz app without
pressing stop and restart it

let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 21 giu 2024 alle 23:16 Mark Edington ***@***.***>
ha scritto:

**Reply by medington** [reply]
Thanks for the reply Roberto!

> with the current version you can achieve what you need simply pressing
> start and when you are ok with the warmup just close the qz app without
> pressing stop and restart it

 💡 - I should have thought of that! 🤦🏼‍♂️ That totally works of course.

Still, it would be great if the app did just show the wattage without recording if that can be made to work. Having to kill / restart the app at the end of the warmup isn't quite seamless.

If you do add a prompt for the Strava upload I think others might appreciate that, but some might find it annoying so probably best to also add an option in the settings like "Upload to Strava" (Auto, Prompt, Disabled). Happy to pull a branch and build it locally to test and give you feedback before you push to the app store if you would like.

Grazie mille!

--Mark

**Reply by cagnulein** [reply]
@medington i didn't even remember that there is already a "Instant Power on Pause" setting in the advanced setting to calculate the wattage even on pause! Let me know!

**Reply by cagnulein** [reply]
> If you do add a prompt for the Strava upload I think others might appreciate that, but some might find it annoying so probably best to also add an option in the settings like "Upload to Strava" (Auto, Prompt, Disabled). Happy to pull a branch and build it locally to test and give you feedback before you push to the app store if you would like.

done, @medington i will test it tomorrow on android and ios https://github.com/cagnulein/qdomyos-zwift/pull/2406

**Reply by cagnulein** [reply]
just tested and merged with the master @medington

**Comment by cagnulein** [comment]
ok i'm overwhelmed by requests in these days, i will try the best i can
have a nice sunday!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 23 giu 2024 alle 20:11 Mark Edington ***@***.***>
ha scritto:
